crates/biome_service/src/file_handlers/javascript.rs

```diff
@@ -422,6 +397,58 @@ pub(crate) fn lint(params: LintParams) -> LintResults {
                 rule_filter_list.push(RuleFilter::Rule("correctness", "noSuperWithoutExtends"));
                 rule_filter_list.push(RuleFilter::Rule("nursery", "noSuperWithoutExtends"));
                 rule_filter_list
+            } else {
+                // We execute a single rule or group because the `--only` filter is specified.
+                let mut enabled_rules = rustc_hash::FxHashSet::default();
+                let mut only_groups = rustc_hash::FxHashSet::default();
+                let mut skipped_groups = rustc_hash::FxHashSet::default();
+                let mut skipped_rules = rustc_hash::FxHashSet::<RuleFilter>::default();
+                for selector in &params.only {
+                    match selector {
+                        RuleSelector::Group(group) => {
+                            only_groups.insert(group.as_str());
+                        }
+                        RuleSelector::Rule(group, rule_name) => {
+                            if let Some(rules) = rules.as_mut() {
+                                // Set the severity level of the rule to its default.
+                                rules
+                                    .to_mut()
+                                    .set_default_severity_if_off(*group, rule_name);
+                            }
+                            enabled_rules.insert((*selector).into());
+                        }
+                    }
+                }
+                for selector in &params.skip {
+                    match selector {
+                        RuleSelector::Group(group) => {
+                            skipped_groups.insert(group.as_str());
+                        }
+                        RuleSelector::Rule(_, _) => {
+                            skipped_rules.insert((*selector).into());
+                        }
+                    }
+                }
+                if !params.skip.is_empty() || !only_groups.is_empty() {
+                    if let Some(rules) = rules.as_mut() {
+                        // Ensure that the recommended field is not set to `false`.
+                        rules.to_mut().set_recommended();
+                    }
+                    enabled_rules.extend(
+                        rules
+                            .as_ref()
+                            .map(|rules| rules.as_enabled_rules())
+                            .unwrap_or_default()
+                            .into_iter()
+                            .filter(|rule_filter| {
+                                (params.only.is_empty()
+                                    || only_groups.contains(&rule_filter.group()))
+                                    && !skipped_groups.contains(&rule_filter.group())
+                                    && !skipped_rules.contains(rule_filter)
+                            }),
```

Comment:
Since we're using a `HashSet`, why don't we use methods such as `intersection` and `difference`? That's one of the reasons why we picked this data structure.

Also, why don't we compute these rules at `AnalysisFilter` level? The `AnalysisFilter` is already ready for this scenario, in fact it accepts `disabled_rules`

https://github.com/biomejs/biome/blob/a51ef9d771c526c5df42da0167a65813b7e80854/crates/biome_analyze/src/lib.rs#L822-L832